export/httpQuery.ts

```diff
@@ -272,6 +278,12 @@ export function neon(
       'Neon-Array-Mode': 'true', // this saves data and post-processing even if we return objects, not arrays
     };
 
+    if (typeof authToken === 'string') {
+      headers['Authorization'] = `Bearer ${authToken}`;
+    } else if (typeof authToken === 'function') {
+      headers['Authorization'] = `Bearer ${Promise.resolve(authToken())}`;
+    }
```

Comment:
Doesn't seem to handle async authToken functions properly:
![Screenshot 2024-09-24 at 13 31 12](https://github.com/user-attachments/assets/e44a9c88-0934-447e-8bb8-f4ad8f1f0133)

Reply:
Fixed by awaiting for the promise to resolve and also added a few other promise validations.  However, as you can see on the video, we don't allow `null` to be passed, so this needs a little bit more boilerplate on the consumer side. We could probably widen the signature for this use case, but it could look weird to accept `null` 🤔 

e.g.

```ts
    const sql = neon(
      'postgresql://anonymous@ep-fancy-cell-w001y1ab.cloud.nitrogen.aws.neon.build/neondb?sslmode=require',
      {
        authToken: async () => {
          const token = await auth.getToken();
          if (!token) {
            throw new Error('No token');
          }
          return  token;
        },
      }
    );
```


https://github.com/user-attachments/assets/fa6e9d41-58fc-4385-86bd-f01229127618

Reply:
@pffigueiredo is this final now? Just double checking because today we talked about this.

Reply:
Yeah, I think this is the right call, accepting `null`\`undefined` makes no sense in there 👍